query-engine/connectors/mongodb-query-connector/src/orderby.rs

```diff
@@ -105,27 +104,23 @@ impl OrderByData {
 
     /// The Mongo binding name of this orderBy, required for cursor conditions.
     /// For a relation field, this is only the first item of the path (e.g. orderby_TestModel_1)
-    /// Returns 2 forms, one for the left one and one for the right one. The left one is
-    /// escaped for usage in user-defined variables.
+    /// Returns 2 forms, one for the left one and one for the right one. The left one is a unique,
+    /// valid name for a variable binding.
     pub(crate) fn binding_names(&self) -> (String, String) {
         if let Some(ref prefix) = self.prefix {
             let first = prefix.first().unwrap().to_string();
 
             (first.clone(), first)
         } else {
             // TODO: Order by relevance won't work here
-            let right = self
-                .order_by
-                .field()
-                .expect("a field on which to order by is expected")
-                .db_name()
-                .to_owned();
-
-            let left = if right.starts_with('_') {
-                right.strip_prefix('_').unwrap().to_owned()
-            } else {
-                right.to_owned()
-            };
+            let field = self.order_by.field().expect("a field on which to order by is expected");
+            let right = field.db_name().to_owned();
+            let mut left = field.name.clone();
+
+            // Issue: https://github.com/prisma/prisma/issues/14001
+            // Here we can assume the field name is ASCII, because it is the _client_ field name,
+            // not the mapped name.
+            left[..1].make_ascii_lowercase();
```

Comment:
Sorry, I don't understand what `left[..1]` is doing here?

Reply:
It just makes the first character of the `left` String lowercase (it takes a mutable slice of just the first character, then makes the whole slice lowercase). I'll add a comment to make that less obscure.

Reply:
ok thanks.